HLTrigger/JetMET/plugins/HLTHFJetCleaner.cc

```diff
@@ -0,0 +1,268 @@
+// -*- C++ -*-
+//
+// Package:    HLTrigger/HLTHFJetCleaner
+// Class:      HLTHFJetCleaner
+//
+/**\class HLTHFJetCleaner HLTHFJetCleaner.cc HLTrigger/HLTHFJetCleaner/plugins/HLTHFJetCleaner.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Alp Akpinar
+//         Created:  Wed, 01 Dec 2021 15:47:19 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/JetReco/interface/PFJet.h"
+#include "DataFormats/METReco/interface/CaloMET.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+//
+// class declaration
+//
+
+class HLTHFJetCleaner : public edm::stream::EDProducer<> {
+public:
+  typedef std::vector<reco::PFJet> JetCollection;
+  explicit HLTHFJetCleaner(const edm::ParameterSet&);
+  ~HLTHFJetCleaner() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void beginStream(edm::StreamID) override;
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  void endStream() override;
+
+  template <typename T>
+  std::vector<T> extractValueMapParameters(edm::Handle<edm::ValueMap<T>> valueMapHandle, size_t numJets);
+
+  edm::EDGetTokenT<edm::View<reco::PFJet>> jetToken_;
+  edm::EDGetTokenT<edm::View<reco::CaloMET>> metToken_;
+
+  edm::EDGetTokenT<edm::ValueMap<float>> sigmaEtaEtaToken_;
+  edm::EDGetTokenT<edm::ValueMap<float>> sigmaPhiPhiToken_;
+  edm::EDGetTokenT<edm::ValueMap<int>> centralEtaStripSizeToken_;
+
+  // Minimum jet pt to consider while applying HF cuts
+  float jetPtThresh_;
+  float dphiJetMetThresh_;
+
+  // Cuts will be applied to jets only with jetEtaMin_ < |eta| < jetEtaMax_
+  float jetEtaMin_;
+  float jetEtaMax_;
+
+  float sigmaEtaPhiDiffThresh_;
+  float centralEtaStripSizeThresh_;
+  bool applyCornerCut_;
+  bool applySigmaEtaPhiCut_;
+  bool applyStripSizeCut_;
```

Comment:
Make class data members `const` wherever possible.

Reply:
Implemented in [fab4d48](https://github.com/cms-sw/cmssw/pull/37539/commits/fab4d48424a5f8f5544e8a4cc678fc5ce9167998)